pkg/snapshotmgr/snapshot_manager.go

```diff
@@ -803,6 +816,31 @@ func uploadCRNameForSnapshotPEID(snapshotPEID astrolabe.ProtectedEntityID) (stri
 	return "upload-" + snapshotPEID.GetSnapshotID().String(), nil
 }
 
+func decodePeIdFromSnapshotID(snapshotID astrolabe.ProtectedEntitySnapshotID, logger logrus.FieldLogger) (astrolabe.ProtectedEntityID, error) {
+	var peId astrolabe.ProtectedEntityID
+	snapshotID64Str := snapshotID.String()
+	snapshotIDBytes, err := base64.RawStdEncoding.DecodeString(snapshotID64Str)
+	if err != nil {
+		errorMsg := fmt.Sprintf("decodePeIdFromSnapshotID: Could not decode snapshot ID encoded string %s", snapshotID64Str)
+		logger.WithError(err).Error(errorMsg)
+		return peId, errors.Wrap(err, errorMsg)
+	}
+	peId, err = astrolabe.NewProtectedEntityIDFromString(string(snapshotIDBytes))
+	if err != nil {
+		errorMsg := fmt.Sprintf("decodePeIdFromSnapshotID: Could not translate decoded snapshotID %s into pe-id", string(snapshotIDBytes))
+		logger.WithError(err).Error(errorMsg)
+		return peId, errors.Wrap(err, errorMsg)
+	}
+	logger.Infof("decodePeIdFromSnapshotID: Successfully translated snapshotID %s into pe-id: %s", snapshotID.String(), peId.String())
```

Comment:
If I remember correctly, the same code snippet is also used in the astrolabe part of deleteSnapshot fix. It would be better to create a astrolabe util function for it, so that both astrolabe PVC PE and vsphere plugin can consume it.

Reply:
Actually, they are different, astrolabe method does both encode and decode depending on the conditions, this one only does a decode.